frontend/src/features/admin-form/settings/components/MultiLanguageSection/FormMultiLanguageToggle.tsx

```diff
@@ -0,0 +1,202 @@
+              aria-label={`Select ${unicodeLocale} as the form's default language`}
+              onClick={() => handleToggleSupportedLanague(unicodeLocale)}
+            />
+            <IconButton
+              variant="clear"
+              icon={<BiEditAlt width="44px" />}
+              colorScheme="secondary"
+              aria-label={`Add ${unicodeLocale} translations`}
+              // TODO: Will add redirection to translation section in next PR
+              onClick={() => console.log('Edit translations')}
+            />
+          </HStack>
+        )}
+      </Flex>
+      {!isLast && <Divider />}
+    </>
+  )
+}
+
+const LanguageTranslationSection = ({
+  defaultLanguage,
+}: LanguageTranslationSectionProps): JSX.Element => {
+  let unicodeLocales = Object.values(Language)
+
+  const idxToRemove = unicodeLocales.indexOf(defaultLanguage)
+  unicodeLocales.splice(idxToRemove, 1)
+
+  unicodeLocales = [defaultLanguage, ...unicodeLocales]
+
+  return (
+    <Box pt={8}>
+      {unicodeLocales.map((unicodeLocale, id, arr) => {
+        return (
+          <LanguageTranslationRow
+            unicodeLocale={unicodeLocale}
+            isDefaultLanguage={unicodeLocale === defaultLanguage}
+            isLast={id === arr.length - 1}
+            key={unicodeLocale}
+          />
+        )
+      })}
+    </Box>
+  )
+}
+
+export const FormMultiLanguageToggle = (): JSX.Element => {
+  const { data: settings, isLoading: isLoadingSettings } =
+    useAdminFormSettings()
+
+  const { mutateFormSupportedLanguages, mutateFormHasMultiLang } =
+    useMutateFormSettings()
+
+  const hasMultiLang = useMemo(
+    () => settings && settings?.hasMultiLang,
+    [settings],
+  )
+
+  const handleToggleMultiLang = useCallback(() => {
+    if (!settings || isLoadingSettings || mutateFormHasMultiLang.isLoading)
+      return
+
+    const nextHasMultiLang = !settings.hasMultiLang
+
+    const currentSupportedLanguages = settings?.supportedLanguages
+
+    // Restore back previously supported languages.
+    let nextSupportedLanguages = currentSupportedLanguages
+
+    // Add all languages as this is the first instance user turns on
+    // toggle
+    if (_.isEmpty(currentSupportedLanguages) && nextHasMultiLang) {
+      nextSupportedLanguages = Object.values(Language)
+    }
+
+    mutateFormSupportedLanguages.mutate({ nextSupportedLanguages })
+
+    return mutateFormHasMultiLang.mutate(nextHasMultiLang)
+  }, [
+    isLoadingSettings,
+    mutateFormHasMultiLang,
+    mutateFormSupportedLanguages,
+    settings,
+  ])
+
+  return (
+    <Skeleton isLoaded={!isLoadingSettings && !!settings} mt="2rem">
+      <Toggle
+        onChange={handleToggleMultiLang}
+        isChecked={hasMultiLang}
+        label="Enable multi-language"
+        description="This will allow respondents to select a language they prefer to view your form in. Translations are not automated."
+      />
+      {settings && hasMultiLang && (
```

Comment:
same for the convention of doing `{settings && hasMultiLang ? ... : null}` instead of nullish coalesce operator